src/overlays/actors/ovl_Obj_Syokudai/z_obj_syokudai.c

```diff
@@ -22,38 +32,283 @@ const ActorInit Obj_Syokudai_InitVars = {
+    ObjSyokudai* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->standCollider);
+    Collider_DestroyCylinder(globalCtx, &this->flameCollider);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void ObjSyokudai_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjSyokudai* this = THIS;
+    s32 groupSize = OBJ_SYOKUDAI_GET_GROUP_SIZE(thisx);
+    s32 switchFlag = OBJ_SYOKUDAI_GET_SWITCH_FLAG(thisx);
+    s32 type = OBJ_SYOKUDAI_GET_TYPE(thisx);
+    s32 pad0;
+    WaterBox* waterBox;
+    f32 waterSurface;
+    s32 lightRadius = -1;
+    u8 lightIntensity = 0;
+    Player* player;
+    s32 pad1;
+
+    if (this->pendingAction != OBJ_SYOKUDAI_PENDING_ACTION_NONE) {
+        if (ActorCutscene_GetCurrentIndex() != thisx->cutscene) {
+            if (ActorCutscene_GetCanPlayNext(thisx->cutscene) != 0) {
+                ActorCutscene_StartAndSetUnkLinkFields(thisx->cutscene, thisx);
+                if (this->pendingAction > OBJ_SYOKUDAI_PENDING_ACTION_NONE) {
+                    Actor_SetSwitchFlag(globalCtx, switchFlag);
+                }
+            } else {
+                ActorCutscene_SetIntentToPlay(thisx->cutscene);
+            }
+        } else if (func_800F22C4(thisx->cutscene, thisx) != 0) {
+            this->litTimer = -1;
+            this->pendingAction = OBJ_SYOKUDAI_PENDING_ACTION_NONE;
+        }
+    } else {
+        if (func_800CA1E8(globalCtx, &globalCtx->colCtx, thisx->world.pos.x, thisx->world.pos.z, &waterSurface,
+                          &waterBox) &&
+            ((waterSurface - thisx->world.pos.y) > 52.0f)) {
+
+            this->litTimer = 0;
+            if (type == OBJ_SYOKUDAI_TYPE_FLAME_CAUSES_SWITCH) {
+                Actor_UnsetSwitchFlag(globalCtx, switchFlag);
+                if (groupSize != 0) {
+                    this->litTimer = 1;
+                }
+            }
+        } else {
+            s32 interactionType = 0;
+            u32 flameColliderHurtboxDmgFlags = 0;
+            player = PLAYER;
+
+            if (OBJ_SYOKUDAI_GET_START_LIT(thisx)) {
+                this->litTimer = -1;
+            }
+            if (groupSize != 0) {
+                if (Flags_GetSwitch(globalCtx, switchFlag)) {
+                    if (this->litTimer == 0) {
+                        if (type != OBJ_SYOKUDAI_TYPE_SWITCH_CAUSES_FLAME) {
+                            this->litTimer = -1;
+                        } else {
+                            this->pendingAction = OBJ_SYOKUDAI_PENDING_ACTION_CUTSCENE_NO_SWITCH;
+                        }
+                    } else if (this->litTimer > 0) {
+                        this->litTimer = -1;
+                    }
+                } else if (this->litTimer < 0) {
+                    this->litTimer = 20;
+                }
+            }
+            if (this->flameCollider.base.acFlags & AC_HARD) {
```

Comment:
```suggestion
            if (this->flameCollider.base.acFlags & AC_HIT) {
```
I made a dumb dumb in my review